app/src/main/java/com/dkanada/gramophone/ui/activities/LoginActivity.java

```diff
@@ -106,20 +107,21 @@ public void onError(Exception exception) {
         }
     }
 
-    public void check(Context context, String server, String user, String token) {
+    private void check(String server, String user, String token) {
         App.getApiClient().GetSystemInfoAsync(new Response<SystemInfo>() {
             @Override
             public void onResponse(SystemInfo result) {
-                if (Integer.parseInt(result.getVersion().substring(0, 1)) == 1) {
-                    PreferenceUtil.getInstance(context).setServer(server);
-                    PreferenceUtil.getInstance(context).setUser(user);
-                    PreferenceUtil.getInstance(context).setToken(token);
-
-                    Intent intent = new Intent(context, MainActivity.class);
-                    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-                    context.startActivity(intent);
+                if (result.getVersion().charAt(0) == '1') {
+                    PreferenceUtil.getInstance(LoginActivity.this).setServer(server);
+                    PreferenceUtil.getInstance(LoginActivity.this).setUser(user);
+                    PreferenceUtil.getInstance(LoginActivity.this).setToken(token);
+
+                    Intent intent = new Intent(LoginActivity.this, MainActivity.class);
+                    intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+                    startActivity(intent);
+                    finish();
                 } else {
-                    Toast.makeText(context, context.getResources().getString(R.string.error_version), Toast.LENGTH_SHORT).show();
+                    Toast.makeText(LoginActivity.this, getResources().getString(R.string.error_version), Toast.LENGTH_SHORT).show();
```

Comment:
This is interesting, what's the benefit of `LoginActivity.class` over the reference?

Reply:
You save an extra reference. The `this` itself would be different at that place, but with the classname prefix you can specify which `this` you mean. In Kotlin, there's a similar construct where you write `this@LoginActivity`. The key advantage is better readability/clarity and less code.